### PR TITLE
[Backup] `az backup vault resource-guard-mapping`: Add Multiple User Authentication support

### PR DESCRIPTION
**Pull request by akshayneema**
**Related command**
 -  `az backup vault resource-guard-mapping`: Add support for updating, showing, and deleting ResourceGuardProxy
 -  Add Multiple User Authentication (MUA) support for critical operations like:
    -  `az backup vault backup-properties set`: Disable Soft Delete or Security Feature
    -  `az backup item set-policy`: Updating policy associated with a backed-up item
    -  `az backup policy set`: Editing policy to lower retention period
    -  `az backup protection disable`: Deleting a backed-up item
 -  Add `--tenant-id` parameter in critical commands for cross-tenant scenario.

**Description**

**Testing Guide**

**History Notes**

[Backup] `az backup vault resource-guard-mapping`: Add support for updating, showing, and deleting ResourceGuardProxy
[Backup] Add multiple user authentication (MUA) support for critical operations: `az backup vault backup-properties set`/`az backup item set-policy`/`az backup policy set`/`az backup protection disable`
[Backup] Add `--tenant-id` parameter in critical commands: `az backup vault backup-properties set`/`az backup item set-policy`/`az backup policy set`/`az backup protection disable`/`az backup vault resource-guard-mapping` for cross-tenant scenario

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Backup

**Comment by wangzelin007** [comment]
Hi [akshayneema](https://github.com/akshayneema),
May I aks did you test `--tenant-id` parameter in critical commands for cross-tenant scenario？
Others LGTM.

**Comment by wangzelin007** [comment]
> > Hi [akshayneema](https://github.com/akshayneema), May I aks did you test `--tenant-id` parameter in critical commands for cross-tenant scenario？ Others LGTM.
> 
> Hey! Yes, we did a complete manual checking for all the commands including `--tenant-id` parameter. Could not include it in test cases as we do not have a Cross Tenant testing setup in place. Will add some tests in the future if and when we have the setup.
> 
> Also `--tenant-id` is an optional parameter so it should not affect the normal functioning of the command.

Hi [akshayneema](https://github.com/akshayneema),
I updated the history note that will record in the release note.
Please take a look.
image

**Review by wangzelin007 — APPROVED** [review]

**Comment by akshayneema** [comment]
> > > Hi [akshayneema](https://github.com/akshayneema), May I aks did you test `--tenant-id` parameter in critical commands for cross-tenant scenario？ Others LGTM.
> > 
> > 
> > Hey! Yes, we did a complete manual checking for all the commands including `--tenant-id` parameter. Could not include it in test cases as we do not have a Cross Tenant testing setup in place. Will add some tests in the future if and when we have the setup.
> > Also `--tenant-id` is an optional parameter so it should not affect the normal functioning of the command.
> 
> Hi [akshayneema](https://github.com/akshayneema), I updated the history note that will record in the release note. Please take a look. image

Looks good to me @wangzelin007. Thanks for doing this.

File: src/azure-cli/azure/cli/command_modules/backup/_help.py [hunk 1]
```diff
@@ -543,3 +543,28 @@
   - name: Show details of encryption properties of a Recovery Services Vault.
     text: az backup vault encryption show --resource-group MyResourceGroup --name MyVault
 """
+helps['backup vault resource-guard-mapping'] = """
+type: group
+short-summary: Resource Guard Mapping details of the Recovery Services vault.
+"""
+helps['backup vault resource-guard-mapping update'] = """
+type: command
+short-summary: Creates/Updates resource guard mapping of the Recovery Services vault.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_help.py:552]:
```suggestion
short-summary: Create/Update resource guard mapping of the Recovery Services vault.
```

File: src/azure-cli/azure/cli/command_modules/backup/_help.py [hunk 2]
```diff
@@ -543,3 +543,28 @@
...
+examples:
+  - name: Creates/Updates resource guard mapping of the Recovery Services vault.
+    text: az backup vault resource-guard-mapping update --resource-group MyResourceGroup --name MyVault --resource-guard-id MyResourceGuardId
+"""
+helps['backup vault resource-guard-mapping show'] = """
+type: command
+short-summary: Gets resource guard mapping of the Recovery Services vault.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_help.py:559]:
```suggestion
short-summary: Get resource guard mapping of the Recovery Services vault.
```

File: src/azure-cli/azure/cli/command_modules/backup/_help.py [hunk 3]
```diff
@@ -543,3 +543,28 @@
...
+examples:
+  - name: Gets resource guard mapping of the Recovery Services vault.
+    text: az backup vault resource-guard-mapping show --resource-group MyResourceGroup --name MyVault
+"""
+helps['backup vault resource-guard-mapping delete'] = """
+type: command
+short-summary: Deletes resource guard mapping of the Recovery Services vault.
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/_help.py:566]:
```suggestion
short-summary: Delete resource guard mapping of the Recovery Services vault.
```

File: src/azure-cli/azure/cli/command_modules/backup/commands.py [hunk 4]
```diff
@@ -41,6 +41,11 @@ def load_command_table(self, _):
         g.custom_command('encryption update', 'update_encryption')
         g.custom_command('encryption show', 'show_encryption', client_factory=backup_resource_encryption_config_cf)
 
+    with self.command_group('backup vault resource-guard-mapping', backup_custom, client_factory=resource_guard_proxy_cf, exception_handler=backup_exception_handler) as g:
+        g.show_command('update', 'update_resource_guard_mapping')
+        g.show_command('show', 'show_resource_guard_mapping')
+        g.show_command('delete', 'delete_resource_guard_mapping')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/backup/commands.py:47]: Do we need to add `confirmation=True` for the delete command?

Comment (akshayneema) [reply src/azure-cli/azure/cli/command_modules/backup/commands.py:47]: Let me check with the PM and come back.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**